budgeteer-web-interface/src/main/java/org/wickedsource/budgeteer/web/pages/contract/edit/form/EditContractForm.java

```diff
@@ -113,15 +118,23 @@ protected void onSubmit(AjaxRequestTarget target, Form<?> form) {
                 }
             }
         };
+        addAttribute.setOutputMarkupId(true);
         add(addAttribute);
-        add(new AjaxButton("save") {
+        add(new AjaxButton("save", new StringResourceModel(sumbitButtonTextKey)) {
             @Override
             protected void onSubmit(AjaxRequestTarget target, Form<?> form) {
                 try {
                     ((ContractBaseData) form.getModelObject()).getFileModel().setFile(fileUpload.getFile());
                     ((ContractBaseData) form.getModelObject()).getFileModel().setFileName(fileUpload.getFileName());
                     ((ContractBaseData) form.getModelObject()).setContractId(service.save((ContractBaseData) form.getModelObject()));
-                    this.success(getString("feedback.success"));
+                    if(sumbitButtonTextKey.equals("button.save.createmode")){
```

Comment:
This (`EditContractForm`) uses a different approach to `EditBudgetForm` to distinguish between the 2 modes. I personally like the first one using a boolean variable better and think this should be consistent

Reply:
Looking at the code more precisely, I see now that this is not so easy as I thought. I made a new commit refactoring the constructors but left this section unchanged. We can keep it this way for now